props/proptypes.md

```diff
@@ -0,0 +1,226 @@
+| `string`    | A String                                                                                                                                                                                    |
+| `array`     | An Array                                                                                                                                                                                    |
+| `func`      | A function                                                                                                                                                                                  |
+| `number`    | Any numeric value                                                                                                                                                                           |
+| `object`    | A JavaScript object                                                                                                                                                                         |
+| `element`   | A React element like `<img />`                                                                                                                                                              |
+| `oneOf`     | Ensures that the prop is one of a list of specific values. Ex: `name: PropTypes.oneOf(['Ada', 'Soo-ah', 'Chrissy'])` ensures that the `name` prop can only be one of the given values.      |
+| `oneOfType` | Ensures that the prop is one of a list of specific types. Ex: `name: PropTypes.oneOfType([PropTypes.string, PropTypes.number)` ensures that the `name` prop can only be a string or number. |
+| `arrayOf`   | An array of a specific type. Ex: `ages: PropTypes.arrayOf(PropTypes.number)` would only accept an array of numbers.                                                                         |
+| `objectOf`  | An object with fields whose values can only be a specific type. Ex: `courses: PropTypes.objectOf(PropTypes.string),`                                                                        |
+| `shape`     | An object with specific fields and types. Ex: `student: PropTypes.shape({fullName: PropTypes.string, age: PropTypes.number})`                                                               |
+
+## PropTypes in Sofia's `Student` Component
+
+Sofia can define PropTypes validators for the `name` and `email` props in her `Student` component.
+
+She'll define:
+
+- `name` is a string, and it is required
+- `email` is a string, and it is required
+
+```js
+import PropTypes from "prop-types";
+
+const Student = (props) => {
+  return (
+    <li>
+      <ul>
+        <li>Nickname: {props.name}</li>
+        <li>Email: {props.email}</li>
+      </ul>
+    </li>
+  );
+};
+
+Student.propTypes = {
+  name: PropTypes.string.isRequired,
+  email: PropTypes.string.isRequired,
+};
+
+export default Student;
+```
+
+If Sofia's array of student data isn't valid, and there's missing information...
+
+```js
+[
```

Comment:
tie into the current code (I also feel better about the trailing ; if we're actually terminating a statement)
```suggestion
const studentData = [
```